packages/persistent-merkle-tree/src/proof/compactMulti.ts

```diff
@@ -0,0 +1,151 @@
+import {convertGindexToBitstring, Gindex, GindexBitstring} from "../gindex";
+import {BranchNode, LeafNode, Node} from "../node";
+import {computeProofBitstrings} from "./util";
+
+export function computeDescriptor(indices: Gindex[]): Uint8Array {
+  // include all helper indices
+  const proofBitstrings = new Set<GindexBitstring>();
+  const pathBitstrings = new Set<GindexBitstring>();
+  for (const leafIndex of indices) {
+    const leafBitstring = convertGindexToBitstring(leafIndex);
+    proofBitstrings.add(leafBitstring);
+    const {branch, path} = computeProofBitstrings(leafBitstring);
+    path.delete(leafBitstring);
+    for (const pathIndex of path) {
+      pathBitstrings.add(pathIndex);
+    }
+    for (const branchIndex of branch) {
+      proofBitstrings.add(branchIndex);
+    }
+  }
+  for (const pathIndex of pathBitstrings) {
+    proofBitstrings.delete(pathIndex);
+  }
+
+  // sort gindex bitstrings in-order
+  const allBitstringsSorted = Array.from(proofBitstrings).sort((a, b) => a.localeCompare(b));
+
+  // convert gindex bitstrings into descriptor bitstring
+  let descriptorBitstring = "";
+  for (const gindexBitstring of allBitstringsSorted) {
+    for (let i = 0; i < gindexBitstring.length; i++) {
+      if (gindexBitstring[gindexBitstring.length - 1 - i] === "1") {
+        descriptorBitstring += "1".padStart(i + 1, "0");
+        break;
+      }
+    }
+  }
+
+  // append zero bits to byte-alignt
+  if (descriptorBitstring.length % 8 != 0) {
+    descriptorBitstring = descriptorBitstring.padEnd(
+      8 - (descriptorBitstring.length % 8) + descriptorBitstring.length,
+      "0"
+    );
+  }
+
+  // convert descriptor bitstring to bytes
+  const descriptor = new Uint8Array(descriptorBitstring.length / 8);
+  for (let i = 0; i < descriptor.length; i++) {
+    descriptor[i] = Number("0b" + descriptorBitstring.substring(i * 8, (i + 1) * 8));
+  }
+  return descriptor;
+}
+
+function getBit(bitlist: Uint8Array, bitIndex: number): boolean {
+  const bit = bitIndex % 8;
+  const byteIdx = Math.floor(bitIndex / 8);
+  const byte = bitlist[byteIdx];
+  switch (bit) {
+    case 0:
+      return Boolean(byte & 0b1000_0000);
```

Comment:
Should be equivalent to
```suggestion
      return byte & 0b1000_0000 !== 0
```
? Type constructors are always a bit sus